sklearn/preprocessing/_target_encoder.py

```diff
@@ -234,24 +235,45 @@ def fit_transform(self, X, y):
                 self.cv, shuffle=self.shuffle, random_state=self.random_state
             )
 
+        y_1d = y
+        n_classes = None
         X_out = np.empty_like(X_ordinal, dtype=np.float64)
         X_unknown_mask = ~X_known_mask
-
-        for train_idx, test_idx in cv.split(X, y):
+        X_ordinal_trans = X_ordinal
+        if self.target_type_ == "multiclass":
+            y_1d = self._label_binarizer_.inverse_transform(y)
```

Comment:
`y_1d` is the original `y`. I think we can keep using the original `y`:

```python
X_ordinal, X_known_mask, y_encoded, n_categories = self._fit_encodings_all(X, y)
```

This way `y` can be used in `cv.split` and `y_encoded` is used for encoding.

With this, I do not think we need `_label_binarizer_` anymore. We just need to store `n_classes`.

Reply:
Thanks, don't know why I didn't realise this. Cleaner!